src/libs/kata-types/src/config/hypervisor/mod.rs

```diff
@@ -289,34 +289,38 @@ impl CpuInfo {
     pub fn adjust_config(&mut self) -> Result<()> {
         let features: Vec<&str> = self.cpu_features.split(',').map(|v| v.trim()).collect();
         self.cpu_features = features.join(",");
-        Ok(())
-    }
-
-    /// Validate the configuration information.
-    pub fn validate(&self) -> Result<()> {
-        Ok(())
-    }
 
-    /// Get default number of guest vCPUs.
-    pub fn get_default_vcpus(&self) -> u32 {
         let cpus = num_cpus::get() as u32;
+
+        // adjust default_maxvcpus
+        if self.default_maxvcpus == 0 || self.default_maxvcpus > cpus {
```

Comment:
What if the cpus we get from the host is larger than the vcpu number that the VMM can support (e.g Dragonball could only support up to 254 cpus and some AMD MILAN chip could have 256 cpus)

Reply:
This will handle your case I think:

https://github.com/kata-containers/kata-containers/blob/cb7f9524be3247ca08cdbbce0ada2d493afcc7ef/src/libs/kata-types/src/config/hypervisor/dragonball.rs#L64-L66